base/src/main/java/org/aya/core/term/Term.java

```diff
@@ -221,4 +222,29 @@ public static boolean checkSubst(@NotNull SeqLike<@NotNull Param> params, @NotNu
   default TermView view() {
     return () -> this;
   }
+
+  default @NotNull ImmutableSeq<Meta> allMetas() {
+    return new TermOps() {
+      final MutableList<Meta> metas = MutableList.create();
+
+      @Override public @NotNull TermView view() {
+        return Term.this.view();
+      }
+
+      @Override public Term pre(Term term) {
```

Comment:
`TermView` and `TermOps` should also be annotated with `@NotNull` or `@Nullable`, which is addressed in my HEAD. 😄